Clock.py

```diff
@@ -30,61 +30,134 @@
+        elif hours == 12:
+            msg += "noon" + 12 * "L "
+        else:
+            msg += number_to_text(truncate_hours(hours)) + " o'clock     " + truncate_hours(hours) * "L "
+    elif minutes == 15:
+        msg += "a quarter past " + number_to_text(truncate_hours(hours))
+    elif minutes == 30:
+        msg += "half past " + number_to_text(truncate_hours(hours))
+    elif minutes == 45:
+        msg += "a quarter to " + number_to_text(truncate_hours(hours + 1))
+    else:
+        # Last resort: just generate the time in hh:mm format
+        msg += "{hour}:{minute:02d}".format(hour=hours, minute=minutes)
+    return msg
+
+#
+# Send a message as morse on the sounder:
+#
 def announce(s, kob, sender):
     for c in s:
         code = sender.encode(c)
         kob.sounder(code)
 
 try:
+    import argparse
     import sys
     import time
     from pykob import config, kob, morse, log
 
-    log.log('Starting Clock')
-
-    PORT = config.Port # serial port for KOB interface
-    WPM = config.Speed  # code speed (words per minute)
-    SOUND = config.Sound # whether to enable computer sound for sounder
-    MSGS = [
-        ( 900, 'The time is nine oclock     L  L  L  L  L  L  L  L  L'),
-        (1000, 'The time is ten oclock     L  L  L  L  L  L  L  L  L  L'),
-        (1100, 'The time is eleven oclock     L  L  L  L  L  L  L  L  L  L  L'),
-        (1200, 'The time is twelve oclock     L  L  L  L  L  L  L  L  L  L  L  L'),
-        (1300, 'The time is one oclock     L'),
-        (1400, 'The time is two oclock     L  L'),
-        (1500, 'The time is three oclock     L  L  L'),
-        (1600, 'The time is four oclock     L  L  L  L'),
-        (1700, 'The time is five oclock     L  L  L  L  L'),
-        (1800, 'The time is six oclock     L  L  L  L  L  L'),
-        (1900, 'The time is seven oclock     L  L  L  L  L  L  L'),
-        (2000, 'The time is eight oclock     L  L  L  L  L  L  L  L'),
-        (2100, 'The time is nine oclock     L  L  L  L  L  L  L  L  L'),
-        (2200, 'The time is ten oclock     L  L  L  L  L  L  L  L  L  L')]
-
-    myKOB = kob.KOB(port=PORT, audio=SOUND)
-    mySender = morse.Sender(WPM)
+    log.log("Starting Clock")
+
+    clock_parser = argparse.ArgumentParser(parents=[config.PortOverride, config.SoundtOverride, config.WPMOverride])
+    clock_parser.add_argument("-b", "--begin", default=900, type=int, help="Beginning of time announcements ", metavar="time", dest="Begin")
+    clock_parser.add_argument("-e", "--end", default=2230, type=int, help="End of time announcements ", metavar="time", dest="End")
+    clock_parser.add_argument("-i", "--interval", default=60, type=int, help="The time announcement interval in minutes", metavar="minutes", dest="Interval")
+    args = clock_parser.parse_args()
+    
+    port = args.Port # serial port for KOB interface
+    speed = args.Speed  # code speed (words per minute)
+    sound = True if args.Sound == 'ON' else False
```

Comment:
Why to you need to do this? `config.Sound` is a boolean.

Reply:
config.Sound is a boolean but args.Sound is a string ('ON' or 'OFF') and this converts 'sound' to be boolean, since even the string 'OFF' is interpreted as 'True' in an 'if' statement etc.  bool('ON') and bool('OFF') are both True'?

Reply:
I would use `strtobool()`.

Reply:
Done.  The latest change actually includes that.  So now it's just

    args = clock_parser.parse_args()
    ...
    sound = strtobool(args.Sound)